test/stdlib/imaxdiv.c

```diff
@@ -15,10 +15,214 @@
+	{0x480258f92fc38de3, 0x2599b52bb0, 0x1ea450c, 0x2879f11a3},
+	{0x5a3257b1114109c9, 0x2978f9f1aa, 0x22cc30a, 0x1317311b25},
+	{0x3c2c319ca8612a65, 0x73fc01ece, 0x84d0088, 0x3165accf5},
+	{0x4f6034e74a16561b, 0x1f29d53707, 0x28c0daa, 0xd88e07075},
+	{0x206665a7072f1cc1, 0xda87e7cea, 0x25f48c1, 0xd3ddb2057},
+	{0x100c559d7db417d7, 0xb907ebbc2, 0x1634188, 0xa2eae16c7},
+	{0x64c5f83691b47cdd, 0x5aced6ebb, 0x11c17fb7, 0x344109030},
+	{0x32a812777eaf7d53, 0x1cb63fe4f, 0x1c3a9675, 0xb113f938},
+	{0x67478d96865ca6b9, 0x142fa03a, 0x51dcb463d, 0x11359ce7},
+	{0x71024e740deb428f, 0x142d3885, 0x599d9edd5, 0x13b1ae6},
+	{0x52c78160b090b655, 0xd02101c6, 0x65d1b205, 0x1c0a0177},
+	{0x16917d5f9fde38b, 0xfb1566c7, 0x17029e0, 0x1bbe166b},
+	{0x74babc1be2ed9c47, 0x22eda9a6, 0x3578b1967, 0x189b247d},
+	{0x7c5cbf2dfc1db6cd, 0x5f09c060, 0x14efd44d4, 0x5210e74d},
+	{0x7c046071c1ac68c3, 0x3696c8e6, 0x24596d86b, 0x26060a1},
+	{0x3f4c0514b0df5e45, 0xf2c3810, 0x42bf84d39, 0x3aa12b5},
+	{0x23fb9839e8358cbd, 0x24deb54, 0xf9d714151, 0xb9c329},
+	{0x2005d5de30015033, 0x47c06db, 0x7240bccba, 0x104d115},
+	{0x67d59c29e076f499, 0x179f009, 0x465554ac22, 0x10b0767},
+	{0x32d2dd34369c836f, 0x13d3fbf, 0x2902f2fb54, 0x7553c3},
+	{0x3960c3c99fdc2235, 0x1b808ba, 0x21618743cd, 0x11e7743},
+	{0x343bad5adfa9726b, 0xeef444, 0x37f58c51a6, 0x3d8a53},
+	{0x7a4aadd7b4e5f191, 0x129c9, 0x6921bb5a2a53, 0x6b66},
+	{0x1b285999316afead, 0x115477, 0x1912cf6611e, 0x801bb},
+	{0x18e6a86b0473a589, 0x50a12, 0x4f0fabc67d4, 0x210a1},
+	{0xffd2ad0e77b73db, 0x146f14, 0xc8500600a3, 0xba1f},
+	{0x76743abdfb91f081, 0xd5888, 0x8e0303c479c, 0x245a1},
+	{0x63cc9c23f0ed0c9d, 0x6c1e5, 0xec4d5841041, 0x38178},
+	{0x7ad70f846e8f1313, 0x7fdf5, 0xf5ecec69bc9, 0x756b6},
+	{0x60de3d71574eb279, 0x6ea3, 0xe02421997a61, 0x18b6},
+	{0x227f92ef6daab68d, 0x15ec, 0x192dda5d5ed25, 0xf71},
+	{0x2e588bdb751a66bf, 0x229c, 0x156d025c70d97, 0x10bb},
+	{0x60ab67a4e5aeabb, 0x1bf7, 0x374f26f3e3ed, 0x210},
+	{0x224f627be76a8261, 0x4f4, 0x6ed4d3882b567, 0x35},
+	{0x300d1ab91bd0b677, 0xe9c, 0x34a002fb76e63, 0x823},
+	{0x2a63d80e0c52fc7d, 0x32, 0xd90970ebc4383f, 0x2f},
+	{0x2b5dc22562dbe059, 0x30a, 0xe45055015fff5, 0x1c7},
+	{0x4a7fd1078807d52f, 0x18d, 0x300a32f60677d4, 0x16b},
+	{0x41a01ee8ab0849f5, 0x13c, 0x352a3971f57e9d, 0x29},
+	{0x723bacc76bd51551, 0x16, 0x53142091089af83, 0xf},
+	{0x11593d6b3f54de6d, 0x63, 0x2cdc6b1a7f9078, 0x5},
+	{0x756c82d6f7069963, 0x5c, 0x146bea3ba565525, 0x17},
+	{0x6884fa0a8f0c99e5, 0x12, 0x5ce7fab40d6088c, 0xd},
+	{0x5052a2953c528441, 0x7, 0xb7984f0bf79809b, 0x4},
+	{0x58dd1583185ecb57, 0x9, 0x9dfad0e90ee1697, 0x8},
+	{0x4b21d01617167e39, 0x2, 0x2590e80b0b8b3f1c, 0x1},
+
+};
 static void imaxdiv_test(void** state)
 {
-	// TODO imaxdiv not being tested
+	intmax_t x, y, div, mod;
+	size_t i;
+
+	for(i = 0; i < (sizeof(t) / sizeof(*t)); i++)
+	{
+		x = t[i].x;
+		y = t[i].y;
+		div = x / y;
+		mod = x % y;
+		assert_true(div == t[i].div);
+		assert_true(mod == t[i].mod);
+		assert_false(t[i].div == t[i].mod);
```

Comment:
This `assert_false` case is not really necessary - there are valid numerical values where div == mod.